buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -1229,7 +1229,15 @@ public int setBytes(int index, InputStream in, int length)
             if (rootParent.hasArray()) {
                 return rootParent.setBytes(idx(index), in, length);
             }
-            byte[] tmp = ByteBufUtil.threadLocalTempArray(length);
+            final byte[] tmp;
+            // Only make use of ThreadLocal if we use a FastThreadLocalThread to make the implementation
+            // Virtual Thread friendly.
+            // See https://github.com/netty/netty/issues/14609
+            if (Thread.currentThread() instanceof FastThreadLocalThread) {
+                tmp = ByteBufUtil.threadLocalTempArray(length);
```

Comment:
HOW about move this inside the ByteBufUtil？

Reply:
I wanted to keep the change "local" to the new allocator for now... @chrisvest @franz1981 WDYT ?

Reply:
That's a good idea IMO

Reply:
Yeah, I think it makes sense in `ByteBufUtil`

Reply:
Addressed